### PR TITLE
Refactored headers into respective directories

### PR DESCRIPTION
**Pull request by Zyro9922**
To increase readability and reduce confusion between coordinate system, reference, representation, etc.

Files were relocated from **the coordinate folder** to respective directories.

New directories with sub-directories

**1. coordinate_system**
     alt_az.hpp	      
     coordinate.hpp	  
     galactic.hpp	    
     geocentric.hpp	  
     heliocentric.hpp	
     supergalactic.hpp
**2. differentials**
     base_differential.hpp	
     cartesian_differential.hpp	 
     differential.hpp	 
     spherical_coslat_differential.hpp	 
     spherical_differential.hpp	 
     spherical_equatorial_differential.hpp
**3. io**
     io.hpp
**4. reference_frames**
     base_ecliptic_frame.hpp	 
     base_equatorial_frame.hpp	 
     base_frame.hpp	 
     cirs.hpp	 
     frame.hpp	 
     icrs.hpp	
**5. rep**
     base_representation.hpp	 
     cartesian_representation.hpp	 
     representation.hpp	 
     spherical_equatorial_representation.hpp	 
     spherical_representation.hpp

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -142,3 +142,14 @@ enable_testing()
 if(BOOST_ASTRONOMY_BUILD_TEST)
 	add_subdirectory(test)
 endif()
+
+#-----------------------------------------------------------------------------
+# SOFA
```
Comment (lpranam) [inline CMakeLists.txt:147]: This is not acceptable. please remove it for now

Comment (lpranam) [inline CMakeLists.txt]: This is not acceptable. please remove it for now

**Review by lpranam — CHANGES_REQUESTED** [review]

**Comment by lpranam** [comment]
can you please update the description which files are relocated from where to where. So many files are changes and it is hard to figure things out :)

**Comment by Zyro9922** [comment]
Sure. I'll update it in description.

**Pull request closed without merging**